sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -349,19 +349,41 @@ object functions extends LegacyFunctions {
   }
 
   /**
-   * Aggregate function: returns the first value in a group.
-   *
-   * @group agg_funcs
-   * @since 1.3.0
-   */
-  def first(e: Column): Column = withAggregateFunction { new First(e.expr) }
+    * Aggregate function: returns the first value in a group. The function does not consider null
+    * values when the ignoreNulls flag is set to true.
```

Comment:
Can you write something like this to be more clear? And update all the docs (including Python).

"The function by default includes the first value it sees. When ignoreNulls is set to true, then it ignores the null values and includes the first non-null value. If all values are null, then null is returned."